### PR TITLE
some changes in vision / grad cam

### PR DESCRIPTION
**Pull request by YassineYousfi**
👋 keras team! 
Made some changes to the grad cam tutorial. Here is a summary of the changes:

- Changed the grad cam function so that there is no need to list the remaining layers after the desired grad cam layer, this should help when the net's graph is not sequential after the chosen layer, e.g. #391 
- Removed last layer's activation function, c.f. #370 
- Added an option to specify which output neuron we wish to get the grad cam for. If this is not given, the argmax is used,
- Avoided for loop with a matmul,
- Wrapped the overlay heatmap and display in a function for later use,
- Added a "multi-label" example with both a cat and a dog in the same image.

I hope these changes are helpful. Happy to further work on this if needed.

### CODE REVIEW

**Comment by YassineYousfi** [comment]
@googlebot I signed it!

**Review by Jimut123 — APPROVED** [review]
Looks Good

**Review by fchollet — APPROVED** [review]
This is an improvement, thanks for the PR!

Please add the generated files.

File: examples/vision/grad_cam.py [hunk 1]
```diff
@@ -36,21 +36,18 @@
...
 last_conv_layer_name = "block14_sepconv2_act"
-classifier_layer_names = [
-    "avg_pool",
-    "predictions",
-]
 
 # The local path to our target image
 img_path = keras.utils.get_file(
-    "african_elephant.jpg", "https://i.imgur.com/Bvro0YD.png"
+    "african_elephant.jpg", " https://i.imgur.com/Bvro0YD.png"
```
[the remark this answers is not in the document]
Comment (YassineYousfi) [reply examples/vision/grad_cam.py:42]: Will do. Thanks.

**Comment by fchollet** [comment]
Thank you!
